test/thirdparty/catch/catch.hpp

```diff
@@ -2153,7 +2153,7 @@ namespace Catch {
                 expr; \
                 __catchResult.captureResult( Catch::ResultWas::DidntThrowException ); \
             } \
-            catch( exceptionType ) { \
+            catch( exceptionType& ) { \

```

Comment:
There is no need to patch Catch here.

Instead, try if the warning disappears if you replace calls like

```cpp
CHECK_THROWS_AS(m.at("Z"), std::out_of_range);
```

with

```cpp
CHECK_THROWS_AS(m.at("Z"), std::out_of_range&);
```

Reply:
@nlohmann  thank you, I have made corresponding changes